basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/UndertowOperations.java

```diff
@@ -139,153 +73,115 @@ public void removeSecurityRealm(String realmName) throws CommandFailedException,
     }
 
-    public String createBufferPool() throws InterruptedException, TimeoutException, IOException {
-        String name = "UndertowBufferPool_" + RandomStringUtils.randomAlphanumeric(6);
-        ResourceAddress address = ioBufferPoolAddressTemplate.resolve(context, name);
-        executeAddAction(address);
+    public String createSocketBinding(String name) throws IOException, CommandFailedException {
+        int port = AvailablePortFinder.getNextAvailable();
+        log.info("Obtained port for socket binding '" + name + "' is " + port);
+        client.apply(new AddSocketBinding.Builder(name)
+                .port(port)
+                .build());
         return name;
     }
 
-    public void removeBufferPool(String name) throws InterruptedException, TimeoutException, IOException {
-        ResourceAddress address = ioBufferPoolAddressTemplate.resolve(context, name);
-        executeRemoveAction(address);
-    }
-
     public String createSocketBinding() throws CommandFailedException, IOException {
         String name = "UndertowSocketBinding_" + RandomStringUtils.randomAlphanumeric(6);
-        try (OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient()) {
-            int port = AvailablePortFinder.getNextAvailable();
-            log.info("Obtained port for socket binding '" + name + "' is " + port);
-            client.apply(new AddSocketBinding.Builder(name)
-                    .port(port)
-                    .build());
+        return createSocketBinding(name);
+    }
+
+    /**
+     * If keepReference is set to true, name of created socket binding name will be kept internally in UndertowOperations
+     * @param keepReference whether the name of newly created socket binding will be saved
+     */
+    public String createSocketBinding(boolean keepReference) throws IOException, CommandFailedException {
+        String socketBinding = createSocketBinding();
+        if (keepReference) {
+            usedSocketBindings.add(socketBinding);
         }
-        return name;
+        return socketBinding;
     }
 
-    public void removeSocketBinding(String name) throws InterruptedException, TimeoutException, IOException {
-        ResourceAddress address = socketBindingAddressTemplate.resolve(context, name);
-        executeRemoveAction(address);
+    public void removeSocketBinding(String name) throws InterruptedException, TimeoutException, IOException, CommandFailedException {
+        client.apply(new RemoveSocketBinding(name));
     }
 
-    public String addBufferCache() throws IOException {
+    public String createBufferCache() throws IOException {
         String name = "BufferCache_" + RandomStringUtils.randomAlphanumeric(6);
         Address address = Address.subsystem("undertow").and("buffer-cache", name);
         operations.add(address);
         return name;
     }
 
+    public String createBufferCache(boolean keepReference) throws IOException {
```

Comment:
Is keepReference argument necessary? Dou you really expect false value to be passed?